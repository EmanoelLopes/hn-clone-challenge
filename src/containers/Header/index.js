import React, { Component } from 'react';
import { faBars, faPlus } from '@fortawesome/free-solid-svg-icons';
import HeaderWrapper from './styles';
import Logo from '../../components/Logo';
import Button from '../../components/Button';
import Search from '../../components/Search';
import Avatar from '../../components/Avatar';
import Badge from '../../components/Badge';
import Renan from '../../assets/images/renan.jpg';
class Header extends Component {
  render() {
    return (
      <HeaderWrapper>
        <div className="container">
          <div className="app-header-left">
            <Logo />
            <Button default={true} icon={faBars} />
            <Search />
          </div>
          <div className="app-header-right">
            <Button type="primary add-post" icon={faPlus} label="Add Post" />
            <Avatar source={Renan} altText="Renan" hasBadge={true}>
              <Badge amount="9" />
            </Avatar>
          </div>
        </div>
      </HeaderWrapper>
    ); 
  }
};

export default Header;