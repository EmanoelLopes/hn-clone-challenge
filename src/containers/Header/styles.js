import styled from 'styled-components';

const HeaderWrapper = styled.header`
  padding: 1.5rem 0;  
  border-bottom: 1px solid #ddd;

  .container {
    display: flex;
    flex-wrap: wrap;
    padding: 0 1rem;

    @media screen and (min-width: 1000px) {
      padding: 0;
    }
  }

  .container .app-header-left,
  .container .app-header-right {
    display: flex;
    align-items: center;
    width: 50%;

    .search {
      width: 100%;
    }

    .search,
    .add-post {
      display: none;
    }

    @media screen and (min-width: 1000px) {
      .search,
      .add-post {
        display: block;
      }
    }
  }

  .container .app-header-left {
    justify-content: flex-start;
  }

  .container .app-header-right {
    justify-content: flex-end;
  }

  .container .app-header-left .app-logo {
    padding: 0 1rem 0 0;
    margin: 0 1rem 0 0;
    border-right: 1px solid #ddd;

    & > a {
      display: block;
    }
  }

  .container .app-header-right .avatar {
    position: relative;

    .badge {
      position: absolute;
      top: 10px;
      left: 6px;
    }
  }
`;

export default HeaderWrapper;