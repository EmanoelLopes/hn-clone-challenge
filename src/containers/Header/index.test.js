import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../Header';

configure({ adapter: new Adapter() });

describe('<Header />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});

