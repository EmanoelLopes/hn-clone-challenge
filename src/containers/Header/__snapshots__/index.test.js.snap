// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="app-header-left"
        >
          <Logo />
          <Button
            default={true}
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f0c9",
                  "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                ],
                "iconName": "bars",
                "prefix": "fas",
              }
            }
          />
          <Connect(Search) />
        </div>
        <div
          className="app-header-right"
        >
          <Button
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f067",
                  "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                ],
                "iconName": "plus",
                "prefix": "fas",
              }
            }
            label="Add Post"
            type="primary add-post"
          />
          <Avatar
            altText="Renan"
            hasBadge={true}
            source="renan.jpg"
          >
            <Badge
              amount="9"
            />
          </Avatar>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="app-header-left"
          >
            <Logo />
            <Button
              default={true}
              icon={
                Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f0c9",
                    "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                  ],
                  "iconName": "bars",
                  "prefix": "fas",
                }
              }
            />
            <Connect(Search) />
          </div>,
          <div
            className="app-header-right"
          >
            <Button
              icon={
                Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f067",
                    "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                  ],
                  "iconName": "plus",
                  "prefix": "fas",
                }
              }
              label="Add Post"
              type="primary add-post"
            />
            <Avatar
              altText="Renan"
              hasBadge={true}
              source="renan.jpg"
            >
              <Badge
                amount="9"
              />
            </Avatar>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Logo />,
              <Button
                default={true}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f0c9",
                      "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                    ],
                    "iconName": "bars",
                    "prefix": "fas",
                  }
                }
              />,
              <Connect(Search) />,
            ],
            "className": "app-header-left",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "default": true,
                "icon": Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f0c9",
                    "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                  ],
                  "iconName": "bars",
                  "prefix": "fas",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f067",
                      "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                    ],
                    "iconName": "plus",
                    "prefix": "fas",
                  }
                }
                label="Add Post"
                type="primary add-post"
              />,
              <Avatar
                altText="Renan"
                hasBadge={true}
                source="renan.jpg"
              >
                <Badge
                  amount="9"
                />
              </Avatar>,
            ],
            "className": "app-header-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "icon": Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f067",
                    "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                  ],
                  "iconName": "plus",
                  "prefix": "fas",
                },
                "label": "Add Post",
                "type": "primary add-post",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "altText": "Renan",
                "children": <Badge
                  amount="9"
                />,
                "hasBadge": true,
                "source": "renan.jpg",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "amount": "9",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="app-header-left"
          >
            <Logo />
            <Button
              default={true}
              icon={
                Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f0c9",
                    "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                  ],
                  "iconName": "bars",
                  "prefix": "fas",
                }
              }
            />
            <Connect(Search) />
          </div>
          <div
            className="app-header-right"
          >
            <Button
              icon={
                Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f067",
                    "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                  ],
                  "iconName": "plus",
                  "prefix": "fas",
                }
              }
              label="Add Post"
              type="primary add-post"
            />
            <Avatar
              altText="Renan"
              hasBadge={true}
              source="renan.jpg"
            >
              <Badge
                amount="9"
              />
            </Avatar>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="app-header-left"
            >
              <Logo />
              <Button
                default={true}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f0c9",
                      "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                    ],
                    "iconName": "bars",
                    "prefix": "fas",
                  }
                }
              />
              <Connect(Search) />
            </div>,
            <div
              className="app-header-right"
            >
              <Button
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f067",
                      "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                    ],
                    "iconName": "plus",
                    "prefix": "fas",
                  }
                }
                label="Add Post"
                type="primary add-post"
              />
              <Avatar
                altText="Renan"
                hasBadge={true}
                source="renan.jpg"
              >
                <Badge
                  amount="9"
                />
              </Avatar>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Logo />,
                <Button
                  default={true}
                  icon={
                    Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f0c9",
                        "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                      ],
                      "iconName": "bars",
                      "prefix": "fas",
                    }
                  }
                />,
                <Connect(Search) />,
              ],
              "className": "app-header-left",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "default": true,
                  "icon": Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f0c9",
                      "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                    ],
                    "iconName": "bars",
                    "prefix": "fas",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Button
                  icon={
                    Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f067",
                        "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                      ],
                      "iconName": "plus",
                      "prefix": "fas",
                    }
                  }
                  label="Add Post"
                  type="primary add-post"
                />,
                <Avatar
                  altText="Renan"
                  hasBadge={true}
                  source="renan.jpg"
                >
                  <Badge
                    amount="9"
                  />
                </Avatar>,
              ],
              "className": "app-header-right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "icon": Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f067",
                      "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
                    ],
                    "iconName": "plus",
                    "prefix": "fas",
                  },
                  "label": "Add Post",
                  "type": "primary add-post",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "altText": "Renan",
                  "children": <Badge
                    amount="9"
                  />,
                  "hasBadge": true,
                  "source": "renan.jpg",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "amount": "9",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
